Add --save and --style args to analyse_results_GaN

ajjackson opened the pull request:
These can be used with Analysis_Scripts/Styles/pub.mplstyle to make small publication-friendly plots.

I'm not sure how interesting/useful it would be to extend this to the other scripts.

I've also found it helpful to tweak lobsterpy to disable plt.show() if files are being saved, this makes the development loop a bit smoother.

ajjackson committed:
Add --save and --style args to analyse_results_GaN

These can be used with Analysis_Scripts/Styles/pub.mplstyle to make
small publication-friendly plots.

JaGeo commented:
Thank you so much! This looks very helpful! The only other COHP plots that we are showing are the ones from Yb14Mn1Sb11. There, it might be useful as well.

ajjackson committed:
Update plotting stylesheet

A few tweaks for more consistence and control.

The key thing is that autolayout needs to be explicitly set to False
to enusre subplot spacing commands work as expected.

JaGeo commented:
Thanks! Merged!

JaGeo commented:
Currently still getting an error when not using "--save". I think it is connected to the default value. Will try to solve it tomorrow. :)
